jetpack.php

```diff
@@ -79,6 +79,19 @@ function wpcom_vip_log_updating_jetpack_version_option( $option_name, $option_va
 	}
 }
 
+// JP should always be marked as active, even if WPCOM_VIP_JETPACK_LOCAL is not set, b/c site could have it in /plugins
+// but not have it enabled via WPCOM_VIP_JETPACK_LOCAL, which would leave it open to manual activation/deactivation
+global $vip_loaded_plugins;
+
+if ( ! is_array( $vip_loaded_plugins ) ) {
+	$vip_loaded_plugins = array();
+}
+
+// Must include `plugins/` on it here, b/c wpcom_vip_get_filtered_loaded_plugins() filters out without it
+if ( ! in_array( 'plugins/jetpack/jetpack.php', $vip_loaded_plugins, true ) ) {
+	$vip_loaded_plugins[] = 'plugins/jetpack/jetpack.php';
```

Comment:
Do you think we should do a `file_exists` check on this as well to make sure Jetpack is actually in the plugins folder before adding it to the array? On my local test site without Jetpack in the plugins folder I get an error on the plugins screen:

`The plugin jetpack/jetpack.php has been deactivated due to an error: Plugin file does not exist.`

I also wonder if we should check for the presence of `WPCOM_VIP_JETPACK_LOCAL` before running any of this code.

Reply:
Ah, then yeah we should wrap it in `file_exists()` to prevent that warning.

> I also wonder if we should check for the presence of WPCOM_VIP_JETPACK_LOCAL before running any of this code.

I tried to explain this in my inline comment on the code:

```
+// JP should always be marked as active, even if WPCOM_VIP_JETPACK_LOCAL is not set, b/c site could have it in /plugins
 +// but not have it enabled via WPCOM_VIP_JETPACK_LOCAL, which would leave it open to manual activation/deactivation
```

Basically if you add the plugin, but don't set the constant, it'll still show as inactive and allow manual activation.

Reply:
> Basically if you add the plugin, but don't set the constant, it'll still show as inactive and allow manual activation.

Ah of course! As you were :+1: